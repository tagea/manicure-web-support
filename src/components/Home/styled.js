import styled from 'styled-components';
import { Typography } from '@material-ui/core';

export const TopHeaderStyled = styled(Typography)`
  flex-grow: 1;
`;
