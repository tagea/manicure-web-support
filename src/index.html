<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Manicure / Support</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link href="https://fonts.googleapis.com/css?family=Baloo+Bhai|Roboto:300,400" rel="stylesheet">
</head>
<body>

<div id="root" class="root"></div>

<script src="./bundle.js"></script>

</body>
</html>
