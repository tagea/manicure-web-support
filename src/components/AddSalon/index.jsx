import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import { InfoInputs } from './InfoInputs';
import { HoursInputs } from './HoursInputs';
import { PhotosInputs } from './PhotosInputs';

import { FormBlockStyled } from './styled';

const AddSalon = (props) => {
  const { values, changeInput } = props;
  return (
    <Grid container direction="column">
      <FormBlockStyled container spacing={16}>
        <InfoInputs changeInput={changeInput} />
        <Grid item sm={2} />
        <HoursInputs values={values} changeInput={changeInput} />
        <PhotosInputs />
      </FormBlockStyled>
    </Grid>
  );
};

AddSalon.propTypes = {
  values: PropTypes.instanceOf(Object),
  changeInput: PropTypes.func
};

AddSalon.defaultProps = {
  values: {},
  changeInput: () => false
};

export { AddSalon };
