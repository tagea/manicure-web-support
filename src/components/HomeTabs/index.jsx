import React from 'react';
import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Grid from '@material-ui/core/Grid';
import { FabButton } from '../../ui/FabButton';
import LogoutSrc from '../../img/logout.svg';

import { ContainerStyled, TabContainerStyled, AppBarStyled, TabStyled, LogOutButtonStyled } from './styled';

class HomeTabs extends React.Component {
  state = {
    value: 0
  };

  handleChange = (event, value) => {
    this.setState({ value });
  };

  render() {
    const { value } = this.state;
    const { titles, children, handleSubmit } = this.props;

    return (
      <ContainerStyled container justify="center">
        <Grid item md={10}>
          <AppBarStyled position="static" color="default">
            <Tabs value={value} onChange={this.handleChange} indicatorColor="primary" textColor="primary">
              {titles.map((title, i) => <TabStyled key={i} label={title} />)}
            </Tabs>
            <LogOutButtonStyled>
              <FabButton label="Logout" src={LogoutSrc} handleClick={handleSubmit} />
            </LogOutButtonStyled>
          </AppBarStyled>
          { children.map((child, i) => (
            value === i && <TabContainerStyled key={i}>{child}</TabContainerStyled>
          )) }
        </Grid>
      </ContainerStyled>
    );
  }
}

HomeTabs.propTypes = {
  titles: PropTypes.arrayOf(PropTypes.string),
  children: PropTypes.arrayOf(PropTypes.object),
  handleSubmit: PropTypes.func
};

HomeTabs.defaultProps = {
  titles: [''],
  children: [{}],
  handleSubmit: () => false
};

export { HomeTabs };
