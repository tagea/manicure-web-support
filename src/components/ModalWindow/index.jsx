import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import { Button } from '../../ui/Button';

import { GroupButtonStyled, DialogTitleStyled } from './styled';

const ModalWindow = (props) => {
  const { open, titleWindow, closeModal, children, view } = props;
  return (
    <Dialog
      open={open}
      maxWidth="lg"
      fullWidth
    >
      <DialogTitleStyled id="dialog">
        <Grid container justify="space-between">
          <Typography variant="h5">{titleWindow}</Typography>
          <GroupButtonStyled variant="subtitle1">
            <Button handleClick={closeModal} color="default" text="Back" />
            { view
              ? <Button variant="contained" color="secondary" text="Edit" />
              : <Button handleClick={closeModal} variant="contained" color="secondary" text="Save" />
            }
          </GroupButtonStyled>
        </Grid>
      </DialogTitleStyled>
      <DialogContent>
        {children}
      </DialogContent>
    </Dialog>
  );
};

ModalWindow.propTypes = {
  open: PropTypes.bool,
  closeModal: PropTypes.func,
  titleWindow: PropTypes.string,
  children: PropTypes.node,
  view: PropTypes.bool
};

ModalWindow.defaultProps = {
  open: false,
  closeModal: () => false,
  titleWindow: '',
  children: null,
  view: false
};

export { ModalWindow };
