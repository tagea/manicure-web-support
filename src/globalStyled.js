import styled, { createGlobalStyle } from 'styled-components';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { theme } from './ui/Themes';
import bg from './img/background.jpg';

const { palette } = theme;

export const GlobalStyle = createGlobalStyle`
  html{
    display: flex;
    flex-direction: column;
    min-height: 100%;
  }
  body {
    margin: 0;
    display: flex;
    flex-direction: column;
    flex: auto;
    color: ${palette.text.main};
  }
  .root{
    display: flex;
    flex-direction: column;
    flex: auto; 
  }
`;

export const BackgroundStyled = styled(Grid)`
  flex: auto;
  background-color: ${palette.secondary.light};
  background-image: url('${bg}');
  background-size: cover;
  padding-bottom: 40px;
`;

export const HeaderStyled = styled(Typography)`
  &&{
    font-weight: 300;
    font-size: 1.5em;
  }
  padding: 0px;
  padding-bottom: 30px;
`;
