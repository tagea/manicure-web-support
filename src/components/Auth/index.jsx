import React from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import { Logo } from '../../ui/Logo';
import { Input } from '../../ui/Input';
import { Button } from '../../ui/Button';
import { BackgroundStyled } from '../../globalStyled';
import { PaperStyled } from './styled';

const Auth = (props) => {
  const { error, changeInput, pressEnter, handleSubmit } = props;
  return (
    <BackgroundStyled container>
      <Grid container direction="column" justify="center" alignItems="center">
        <Logo />
        <PaperStyled container direction="column" justify="center" alignItems="center">
          <Input
            id="username-support"
            label="Username"
            type="text"
            name="username"
            autoComplete="username"
            size="large"
            error={error}
            onChange={changeInput}
          />
          <Input
            id="password-support"
            label="Password"
            type="password"
            name="password"
            autoComplete="current-password"
            size="large"
            error={error}
            onChange={changeInput}
            onKeyDown={pressEnter}
          />
          <Button variant="contained" size="large" color="secondary" text="Sign in" handleClick={handleSubmit} />
        </PaperStyled>
      </Grid>
    </BackgroundStyled>
  );
};

Auth.propTypes = {
  error: PropTypes.bool,
  handleSubmit: PropTypes.func,
  changeInput: PropTypes.func,
  pressEnter: PropTypes.func
};
Auth.defaultProps = {
  error: false,
  handleSubmit: () => false,
  changeInput: () => false,
  pressEnter: () => false
};

export { Auth };
