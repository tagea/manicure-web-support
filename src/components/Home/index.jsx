import React from 'react';
import PropTypes from 'prop-types';
import { BackgroundStyled } from '../../globalStyled';

const Home = (props) => {
  const { children } = props;
  return (
    <BackgroundStyled direction="column" container alignItems="flex-start">
      { children }
    </BackgroundStyled>
  );
};

Home.propTypes = {
  children: PropTypes.node
};
Home.defaultProps = {
  children: null
};

export { Home };
