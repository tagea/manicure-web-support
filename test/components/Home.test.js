import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../../src/components/Home';
import { BackgroundStyled } from '../../src/globalStyled';

it('render Home component', () => {
  const HomeComponent = shallow(<Home />);
  expect(HomeComponent).toMatchSnapshot();
});

it('Home as wrapper BackgroundStyled should contain value "container"', () => {
  const HomeComponent = shallow(<Home />);
  expect(HomeComponent.find(BackgroundStyled).prop('container')).toBeTruthy();
});

it('BackgroundStyled should contain "alignItems" property with "flex-start" value', () => {
  const HomeComponent = shallow(<Home />);
  expect(HomeComponent.find(BackgroundStyled).prop('alignItems')).toEqual('flex-start');
});

it('BackgroundStyled should contain "direction" property with "column" value', () => {
  const HomeComponent = shallow(<Home />);
  expect(HomeComponent.find(BackgroundStyled).prop('direction')).toEqual('column');
});
