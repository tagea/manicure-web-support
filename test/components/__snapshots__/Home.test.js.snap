// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home component 1`] = `
<globalStyled__BackgroundStyled
  alignItems="flex-start"
  container={true}
  direction="column"
/>
`;
